plutus-core/plutus-core/src/PlutusCore/Name/Unique.hs

```diff
@@ -6,8 +6,23 @@
 {-# LANGUAGE TemplateHaskell        #-}
 {-# LANGUAGE TypeApplications       #-}
 
-module PlutusCore.Name (
-  -- * Types
+{- | Defines the 'Name' type used for identifiers in Plutus Core together with a technique
+ to minimise the cost of 'Name' comparisons.
+
+ A 'Name' is a piece of text used to identify a variable inside the Plutus Core languages.
```

Comment:
Technically, it's the `Unique` that identified a `Name`, `Text` is there just for human readability.